lib/core/state/mongo-server.js

```diff
@@ -0,0 +1,33 @@
+const logger = require('../../helpers/logger')
+const ReplicationCommands = require('../commands/replication-commands')
+
+const REPLICATION_STATUS = {
+  UNKNOWN: 0,
+  UNINITIALIZED: 1
+}
+
+class MongoServerState {
+  constructor(client) {
+    this.client = client
+    this.replicationStatus = REPLICATION_STATUS.UNKNOWN
+  }
+
+  async refreshState() {
+    logger.debug(`${this.client.uri}: Refreshing status`)
+    const replicationCommands = new ReplicationCommands(this.client)
+    let status
+
+    try {
+      status = await replicationCommands.status()
+    } catch (error) {
+      if (!error.code) throw error // Not a MongoError, cannot handle this
+      status = error
+    }
+
+    if (status.code && status.code === 94) {
```

Comment:
Why aren't we just mapping these directly instead of implementing our own numbering scheme again? 

I am aware we need to map them in some way to friendly consts

Reply:
I mapped all of these directly to whaever mongo does